caput/pipeline.py

```diff
@@ -834,8 +1013,28 @@ class TaskBase(config.Reader):
     pipeline yaml file when the pipeline is initialized.  The class attributes
     will be overridden with instance attributes with the same name but with the
     values specified in the pipeline file.
+
+    Attributes
+    ----------
+    broadcast_inputs : bool
+        If true, input queues will be broadcast to process all combinations of
+        entries. Otherwise, items in input queues are removed at equal rate.
+        NOT CURRENTLY IMPLEMENTED
+    limit_outputs : int
+        Limits the number of `next` outputs from this task before finishing.
+        Default is None, allowing an unlimited number of `next` products.
+    base_priority : int
+        Integer priority relative to other tasks. Larger positive values have
+        higher priority, and larger negative numbers have lower priority.
+        Default is 0. After setup, priority will be adjusted based on the
+        difference between number of inputs and number of outputs - tasks
+        with more input than outputs will have increased priority.
```

Comment:
I think some more guidance might be helpful here.

There's two priority ordering systems playing off one another here (the `base_priority` specified by the users and the priority generated by considering inputs/outputs).  You need to explain how these systems interact with each other.

You say here that the priorities get adjusted, but you don't say how.  Possible questions:

* Is there a ballpark number for `base_priority` for a "high priority" task?  Is it 1, 10, 100, 1000?
* If I have four tasks with base priorities of 1, 2, 3, 4.  How likely is it that the tasks end up in a different priority order?  What about if I had four tasks with base priorities of 10000, 20000, 30000, 40000?
* Is the actual value of the base priority important, or only the difference between two priorities?  Or the ratio?  Would a run with tasks with base priorities (1, 2, 3) be the same as one with base priorities (-2, -1, 0), or (10, 20, 30)?

The easiest thing to do might to just have examples to for people to use as guidelines for `base_priority` for typical pipeline tasks.

Reply:
I've updated the documentation to try to be clearer and more explicit. The only thing I'm debating is whether `base_priority` should be an absolute - i.e., the `priority` method returns just the `base_priority` if it's set, or to continue with this method of combining base and dynamic priority

Reply:
The updated description is great.

I think it's better to keep the system as it is, which gives flexibility to use `base_priority` either as an (effectively) absolute priority (by using a large number), or else as a priority hint for the automatic priority calculation (by using a small number).